hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystem.java

```diff
@@ -217,6 +218,9 @@ public void initialize(URI uri, Configuration configuration)
       }
     }
     if (!isNamespaceEnabled && uri.toString().contains(FileSystemUriSchemes.WASB_DNS_PREFIX)) {
+      if (abfsConfiguration.isCpkEnabled()) {
```

Comment:
New config needs to be default false for FNS and default true HNS. As we have kept all our changes honoring FNS or HNS boundaries, this will be a conflict as one of FNS or HNS will have to configure this diff from default now.

How about we just remove the new config and instead at this spot check if abfsConfiguration.getClientProvidedEncryptionKey() is not null or empty ?